### PR TITLE
Azure Pipelines - Added build and release stages

### PR DESCRIPTION
**Pull request by gsvic**
In this PR we added stages in Azure Pipelines. The following were added
- Build the native libraries of OSX, Linux and Windows and publish them as artifacts in Azure Pipelines
- Build a fat jar that contains all native libraries

### CODE REVIEW

File: ci/tiledb-java-fat-jar.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      tarCompression: 'gz' # Optional. Options: gz, bz2, xz, none
+      archiveFile: $(Build.ArtifactStagingDirectory)/tiledb-java-release-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      replaceExistingArchive: true
+      verbose: true # Optional
+    condition: succeeded()
+
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathtoPublish: $(Build.BinariesDirectory)
+      artifactName: fat_jar
```
Comment (Shelnutt2) [inline ci/tiledb-java-fat-jar.yml:71]: Change the artifact name.

File: ci/tiledb-java-linux_osx-release.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      tarCompression: 'gz' # Optional. Options: gz, bz2, xz, none
+      archiveFile: $(Build.ArtifactStagingDirectory)/tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      replaceExistingArchive: true
+      verbose: true # Optional
+    condition: succeeded()
+
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathtoPublish: $(Build.ArtifactStagingDirectory)/tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      artifactName: tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
```
Comment (Shelnutt2) [inline ci/tiledb-java-linux_osx-release.yml:67]: you can use one common artifact name for all libraries. Something like `libraries`.

File: ci/tiledb-java-linux_osx.yml [hunk 3]
```diff
@@ -9,8 +9,9 @@ steps:
   - bash: |
       unset SYSTEM
       ls
-      ./gradlew assemble
+
       ./gradlew checkFormat
+      ./gradlew assemble
```
Comment (Shelnutt2) [inline ci/tiledb-java-linux_osx.yml:14]: Make the format and the assemble separate steps.

File: ci/tiledb-java-windows-release.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      tarCompression: 'gz' # Optional. Options: gz, bz2, xz, none
+      archiveFile: $(Build.ArtifactStagingDirectory)/tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      replaceExistingArchive: true
+      verbose: true # Optional
+    condition: succeeded()
+
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathtoPublish: $(Build.ArtifactStagingDirectory)/tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      artifactName: tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
```
Comment (Shelnutt2) [inline ci/tiledb-java-windows-release.yml:49]: Common artifact name `libraries`.

File: ci/tiledb-java-fat-jar.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      archiveType: 'tar' # Options: zip, 7z, tar, wim
+      tarCompression: 'gz' # Optional. Options: gz, bz2, xz, none
+      archiveFile: $(Build.ArtifactStagingDirectory)/tiledb-java-release-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      replaceExistingArchive: true
+      verbose: true # Optional
+    condition: succeeded()
+
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathtoPublish: $(Build.BinariesDirectory)
```
[the remark this answers is not in the document]
Comment (gsvic) [reply ci/tiledb-java-fat-jar.yml:70]: Shouldn't we upload a single fat jar that contains all native libs?

[the remark this answers is not in the document]
Comment (Shelnutt2) [reply ci/tiledb-java-fat-jar.yml:70]: I mean the doc jar, the source jar and the code jar.

File: ci/tiledb-java-fat-jar.yml (hunk 1 above)
Comment (gsvic) [reply ci/tiledb-java-fat-jar.yml]: Got it. These three are the only files `$(Build.BinariesDirectory)`. Can you check this line? https://github.com/TileDB-Inc/TileDB-Java/pull/171/files#diff-f6cc1790cd29925bdb06907055c4b05bR39

**Pull request merged**